### PR TITLE
Added Check Access Pallet

### PR DESCRIPTION
**Pull request by SouravMishra3256**

### CODE REVIEW

File: check-access/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+] }
+frame-support = { default-features = false, version = "4.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.26"}
+frame-system = { default-features = false, version = "4.0.0-dev", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.26" }
+sp-core = { default-features = false, version = "6.0.0", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.26" }
+sp-io = { default-features = false, version = "6.0.0", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.26" }
+sp-runtime = { default-features = false, version = "6.0.0", git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.26" }
+sp-std = { default-features = false, version = '3.0.0' }
+serde = { features = ['derive'], optional = true, version = '1.0.136' }
+metamui-primitives = { version = "0.0.1", default-features = false, path = '../metamui-primitives'}
+pallet-did = { default-features = false, version = '0.0.1', path = '../did' }
```
Comment (fennsaji) [inline check-access/Cargo.toml:23]: Did pallet needed?

File: check-access/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+		}
+  }
+
+	fn pre_dispatch(
+		self,
+		_: &Self::AccountId,
+		_: &Self::Call,
+		_: &DispatchInfoOf<Self::Call>,
+		_: usize,
+	) -> Result<Self::Pre, TransactionValidityError> {
```
Comment (fennsaji) [inline check-access/src/lib.rs:183]: Add self::validate here, since it was mentioned in comments

File: check-access/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+			Ok(ValidTransaction {
+				priority: info.weight as TransactionPriority,
+				longevity: TransactionLongevity::max_value(),
+				propagate: true,
+				..Default::default()
+			})
+		}
+
+		else{
+			Err(InvalidTransaction::Custom(230).into())
```
Comment (fennsaji) [inline check-access/src/lib.rs:173]: What is 230?

Comment (SouravMishra3256) [reply check-access/src/lib.rs:173]: It was a custom error value to check if the pallet was working

File: check-access/src/lib.rs (hunk 2 above)
Comment (SouravMishra3256) [reply check-access/src/lib.rs:183]: Ok

File: check-access/Cargo.toml (hunk 1 above)
Comment (SouravMishra3256) [reply check-access/Cargo.toml:23]: Added for testing to add DidResolve in Test's Config
But used default `()`  implementation instead

**Review by fennsaji — APPROVED** [review]

File: check-access/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    info: &DispatchInfoOf<Self::Call>,
+    _len: usize,
+  ) -> Result<ValidTransaction, TransactionValidityError> {
+  
+		let pallet_name = <Pallet<T>>::convert_to_array(
+			call.get_call_metadata().pallet_name.as_bytes().to_vec()
+		);
+
+		let function_name = <Pallet<T>>::convert_to_array(
+			call.get_call_metadata().pallet_name.as_bytes().to_vec()
```
Comment (fennsaji) [inline check-access/src/lib.rs:160]: Shouldn't this be function name?

**Review by fennsaji — APPROVED** [review]

**Pull request merged**